docs/content/en/docs/tutorials/ci_cd.md

```diff
@@ -0,0 +1,286 @@
+
+To facilitate building and deploying images in CI/CD skaffold does offer docker images which you can build on top of or if that doesn’t work, you can make your own Dockerfile and pull in skaffold and its dependencies using curl.
+
+Let us have a look at how we can use Skaffold with Gitlab CI.
+
+**Step 1: Getting the project and Dockerfile ready**
+
+The first step is to obviously have your project which you want to deploy ready with a Dockerfile setup for the same. Skaffold supports multiple builders like Docker, Kaniko, Bazel and more (including custom builders). We will use Docker in this demonstration.
+
+For instance, if you would like to deploy nginx, your Dockerfile would look something like this:
+
+
+```
+FROM nginxinc/nginx-unprivileged:1.19.3-alpine
+WORKDIR /app/server
+
+COPY ./web ./web
+
+EXPOSE 8080
+
+CMD [ "/bin/bash", "-c", "nginx -g 'daemon off;'" ]
+```
+
+
+Skaffold can use your dockerfile to automatically build, tag and push images to the registry when needed.
+
+**Step 2: Choosing the deployment method**
+
+Skaffold supports deployment through kubectl, helm and a lot of other mechanisms. You can have a look at the yaml reference for complete list here: [https://skaffold.dev/docs/references/yaml/](https://skaffold.dev/docs/references/yaml/)
+
+So, if you choose to deploy through kubectl you must have all your yaml files ready and if you would like to deploy through helm charts, you must have all your charts ready.
+
+**Step 3: Choosing the image and tag strategy**
+
+Skaffold automatically tags images based on different strategies as documented here: [https://skaffold.dev/docs/pipeline-stages/taggers/](https://skaffold.dev/docs/pipeline-stages/taggers/) and you might want to choose the mechanism which is right for you. By default, it uses the git sha if unspecified.
+
+Also, if you are using helm chart as your deployment method, you might want to use the helm image strategy if you would like to follow helm specific conventions and skaffold will pass in all the details for you.
+
+**Step 4: Plugins (optional)**
+
+Another interesting thing is that skaffold also supports plugins like Helm Secrets ([https://github.com/zendesk/helm-secrets/](https://github.com/zendesk/helm-secrets/)). So, if you would like to deploy secrets that have been encrypted using KMS or any other mechanism supported by SOPS ([https://github.com/mozilla/sops](https://github.com/mozilla/sops)) you can actually set **useHelmSecrets** option to true and skaffold will handle everything automatically for you.
+
+**Step 5: The skaffold.yaml file**
+
+Depending on the way you would like to build, deploy, your skaffold yaml file will actually change and it's quite easy to set up one. It is recommended that you set up skaffold before you setup your CI pipeline since the CI pipeline can just call skaffold to do all the builds and deployments and you can do the same locally as well.
```

Comment:
```suggestion
Your skaffold.yaml file will change depending on the way you would like to build and deploy your project. It is recommended that you set up skaffold before you setup your CI pipeline since the CI pipeline can just call skaffold to do all the builds and deployments, and you can do the same locally as well.
```